src/TSTransformer/nodes/jsx/transformJsxChildren.ts

```diff
@@ -143,7 +136,7 @@ function createJsxAddChild(
 ): luau.Statement {
 	const isPossiblyUndefinedOrFalse = isPossiblyType(type, isUndefinedType, isBooleanLiteralType(state, false));
 	const isPossiblyTrue = isPossiblyType(type, isBooleanLiteralType(state, true));
-	const isPossiblyElement = isPossiblyType(type, isRoactElementType(state));
+	const isPossiblyElement = isPossiblyType(type, isJSXElementType(state));
```

Comment:
These transforms are still very Roact-specific - TS does not modify children values before the JSX factory. Other JSX libraries may very well want to accept `true` as a child element, but this would make it impossible.
Also, issues #2084 and #2139 still remain after these change; That should at least be solved. The failing assert for that is on line 227 before/220 after diff.